### DISCUSSION TITLE
blog-starter - Add example of using <Image /> component inside markdown files

### DISCUSSION THREAD

**Discussion by johncmunson** [vercel/next.js#26856] [Ideas]
### Describe the feature you'd like to request

In the [blog-starter](https://github.com/vercel/next.js/tree/canary/examples/blog-starter) example, there are a few markdown blog posts that get processed and statically rendered as NextJS pages.

Here's a simplified version of what one of those markdown files looks like...
```
---
title: 'Climbing Mt. Everest'
description: 'It was really tall'
coverImage: '/assets/mountains1.jpg'
---

Can't wait to go again next year!
```

As you can see, image paths are being used in the frontmatter. These image paths eventually get handed off to `<Image />` components. However, the blog-starter example unfortunately does not contain any examples of images being used within the body of a markdown file. For example, something like...
```
---
title: 'Climbing Mt. Everest'
description: 'It was really tall'
coverImage: '/assets/mountains1.jpg'
---

Here's a picture of a mountain I climbed

![Mountains](/assets/mountains2.jpg)
```

While this will work, and the image will get rendered to the screen, it will get rendered using the native `<img />` tag instead of the enhanced `<Image />` component.

It would be great to enhance this example project with an example of how to transform markdown into HTML/JSX that uses the `<Image />` component anywhere images are being used within the body of a markdown file.

### Describe the solution you'd like

When the following markdown is transformed to HTML/JSX...
```
![Mountains](/assets/mountains.jpg)
```
It will produce something like this...
```
<Image src="/assets/mountains.jpg" alt="Mountains" />
```
instead of this...
```
<img src="/assets/mountains.jpg" alt="Mountains" />
```

### Describe alternatives you've considered

If this is too difficult or not feasible, I may look into using NextJS with mdx. Perhaps this is something that the mdx folks have already solved out-of-the-box.

**Comment by 0pen-Standards** [comment]
I'd love to see this feature incorporated as well. If anyone has a repo of the above solution for the [blog-starter](https://vercel.com/templates/next.js/blog-starter-kit) template, please share.
Thanks!
